After adding a `cmakeFlags = ["-DBUILD_TESTS=ON"];`
I get this error:
```
[ 80%] Building CXX object test/pytest/src/CMakeFiles/NumCppPy.dir/Rotations.cpp.o
In file included from /build/source/test/pytest/src/pybind11/pybind11/cast.h:15,
                 from /build/source/test/pytest/src/pybind11/pybind11/attr.h:14,
                 from /build/source/test/pytest/src/pybind11/pybind11/detail/class.h:12,
                 from /build/source/test/pytest/src/pybind11/pybind11/pybind11.h:13,
                 from /build/source/test/pytest/src/pybind11/pybind11/complex.h:12,
                 from /build/source/test/pytest/src/BindingsIncludes.hpp:3,
                 from /build/source/test/pytest/src/Misc.cpp:1:
In static member function 'static std::pair<const void*, const pybind11::detail::type_info*> pybind11::detail::type_caster_base<type>::src_and_type(const ityp>
    inlined from 'static pybind11::handle pybind11::detail::type_caster_base<type>::cast(const itype*, pybind11::return_value_policy, pybind11::handle) [with >
    inlined from 'static pybind11::handle pybind11::detail::type_caster_base<type>::cast(const itype&, pybind11::return_value_policy, pybind11::handle) [with >
    inlined from 'pybind11::cpp_function::initialize<pybind11::class_<std::pair<nc::NdArray<unsigned int>, nc::NdArray<unsigned int> > >::def_readonly<std::pa>
    inlined from 'static constexpr pybind11::handle pybind11::cpp_function::initialize<pybind11::class_<std::pair<nc::NdArray<unsigned int>, nc::NdArray<unsig>
/build/source/test/pytest/src/pybind11/pybind11/detail/type_caster_base.h:948:49: error: using a dangling pointer to an unnamed temporary [-Werror=dangling-po>
  948 |         return type_caster_generic::src_and_type(src, cast_type, instance_type);
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/test/pytest/src/pybind11/pybind11/cast.h: In static member function 'static constexpr pybind11::handle pybind11::cpp_function::initialize<pybind>
/build/source/test/pytest/src/pybind11/pybind11/cast.h:1448:51: note: unnamed temporary defined here
 1448 |         return std::forward<Func>(f)(cast_op<Args>(std::move(std::get<Is>(argcasters)))...);
      |                                      ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In static member function 'static std::pair<const void*, const pybind11::detail::type_info*> pybind11::detail::type_caster_base<type>::src_and_type(const ityp>
    inlined from 'static pybind11::handle pybind11::detail::type_caster_base<type>::cast(const itype*, pybind11::return_value_policy, pybind11::handle) [with >
    inlined from 'static pybind11::handle pybind11::detail::type_caster_base<type>::cast(const itype&, pybind11::return_value_policy, pybind11::handle) [with >
    inlined from 'pybind11::cpp_function::initialize<pybind11::class_<std::pair<nc::NdArray<double>, nc::NdArray<double> > >::def_readonly<std::pair<nc::NdArr>
    inlined from 'static constexpr pybind11::handle pybind11::cpp_function::initialize<pybind11::class_<std::pair<nc::NdArray<double>, nc::NdArray<double> > >>
/build/source/test/pytest/src/pybind11/pybind11/detail/type_caster_base.h:948:49: error: using a dangling pointer to an unnamed temporary [-Werror=dangling-po>
  948 |         return type_caster_generic::src_and_type(src, cast_type, instance_type);
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/test/pytest/src/pybind11/pybind11/cast.h: In static member function 'static constexpr pybind11::handle pybind11::cpp_function::initialize<pybind>
/build/source/test/pytest/src/pybind11/pybind11/cast.h:1448:51: note: unnamed temporary defined here
 1448 |         return std::forward<Func>(f)(cast_op<Args>(std::move(std::get<Is>(argcasters)))...);
      |                                      ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Is it possible to not report compiler warnings as errors?

--- a/pkgs/development/libraries/numcpp/default.nix
+++ b/pkgs/development/libraries/numcpp/default.nix
@@ -1,0 +1,48 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  boost,
+  python3,
+  gtest,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "numcpp";
+  version = "2.11.0";
+
+  src = fetchFromGitHub {
+    owner = "dpilger26";
+    repo = "NumCpp";
+    rev = "Version_${finalAttrs.version}";
+    hash = "sha256-IAku1bcaMkawZxpQbvxcS6VX07ogw4UGo1DX2Wa8xwU=";
+  };
+
+  nativeCheckInputs = [gtest python3];
+
+  nativeBuildInputs = [cmake];
+
+  buildInputs = [boost];
+
+  cmakeFlags = lib.optionals finalAttrs.finalPackage.doCheck [
+    "-DBUILD_TESTS=ON"
+    "-DBUILD_MULTIPLE_TEST=ON"
+  ];
+
+  doCheck = !stdenv.isDarwin && !stdenv.hostPlatform.isStatic;
+
+  postInstall = ''
+    substituteInPlace $out/share/NumCpp/cmake/NumCppConfig.cmake \
+      --replace "\''${PACKAGE_PREFIX_DIR}/" ""
+  '';
+
+  NIX_CFLAGS_COMPILE="-Wno-error";
+
+  meta = with lib; {
+    description = "A Templatized Header Only C++ Implementation of the Python NumPy Library";
+    homepage = "https://github.com/dpilger26/NumCpp";
+    license = licenses.mit;
+    maintainers = with maintainers; [spalf];
+    platforms = platforms.unix;
+  };
+})

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -23840,6 +23840,8 @@ with pkgs;
 
   ntrack = callPackage ../development/libraries/ntrack { };
 
+  numcpp = callPackage ../development/libraries/numcpp { };
+
   nuraft = callPackage ../development/libraries/nuraft { };
 
   nuspell = callPackage ../development/libraries/nuspell { };

